tests/integration/test_building_in_subdir.py

```diff
@@ -0,0 +1,33 @@
+"""
+Check that running nikola from subdir works.
+
+Check whether build works from posts/
+"""
+
+import os
+
+import pytest
+
+from nikola import __main__
+
+from .helper import cd
+from .test_demo_build import prepare_demo_site
+from .test_empty_build import (  # NOQA
+    test_archive_exists,
+    test_avoid_double_slash_in_rss,
+    test_check_files,
+    test_check_links,
+    test_index_in_sitemap,
+)
+
+
+@pytest.fixture(scope="module")
+def build(target_dir):
+    """Fill the site with demo content and build it."""
+    prepare_demo_site(target_dir)
+
+    build_dir = os.path.join(target_dir, "posts")
+
+    with cd(build_dir):
+        result = __main__.main(["build"])
+        assert result == 0
```

Comment:
This `assert` does not appear anywhere else, why is it there?

Reply:
It's been like this for ages and I think it has been used to check for a valid build since the build is triggered in a different directory.

Since we are employing more tests now that check for valid builds (and fail if no proper build has been made) I'd say we remove it.
What do you think?

Reply:
If a non-zero exit code will guarantee other tests in this file failing, and if that’s the case for all tests where we build something, then this `assert` can be dropped.